src/components/apply/form/application-form.tsx

```diff
@@ -1,4 +1,133 @@
-// TODO: finish form component
+import type { z } from "zod";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { useForm } from "react-hook-form";
+import {
+  Form,
+  FormControl,
+  FormField,
+  FormItem,
+  FormLabel,
+} from "~/components/ui/form";
+import { Textarea } from "~/components/ui/textarea";
+import { api } from "~/utils/api";
+import { useAutoSave } from "~/components/hooks/use-auto-save";
+import { applicationStepSaveSchema } from "~/schemas/application";
+
 export function ApplicationForm() {
-  return <></>;
+  const utils = api.useUtils();
+  const { data: defaultValues } = api.application.get.useQuery();
+  const { mutate } = api.application.save.useMutation({
+    onSuccess: () => {
+      return utils.application.get.invalidate();
+    },
+  });
+
+  const form = useForm<z.infer<typeof applicationStepSaveSchema>>({
+    resolver: zodResolver(applicationStepSaveSchema),
+    mode: "onBlur",
+  });
+
+  useAutoSave(form, onSubmit, defaultValues);
+
+  function onSubmit(data: z.infer<typeof applicationStepSaveSchema>) {
+    mutate({
+      ...defaultValues,
+      ...data,
+    });
+  }
+
+  const maxChars = 150;
+
+  return (
+    <Form {...form}>
+      <form onSubmit={form.handleSubmit(onSubmit)} className="space-y-4">
+        <div className="flex w-full flex-wrap gap-2">
+          <FormLabel className="w-full">
+            Tell us why you think you will win Hack Western 11.
```

Comment:
Here are the actual application questions

If you could have any superpower to help you during Hack Western, what would it be and why?
If you could build your own dream destination what would it look like? Be as detailed and creative as you want!
What project (anything you have ever worked on not just restricted to tech) of yours are you the most proud of and why? What did you learn throughout the process?